docs/user/key_concepts_overview.md

```diff
@@ -0,0 +1,335 @@
+(key_concepts_overview)=
+
+# Key concepts in atomate2: `Job` & `Flow` `Makers`, `InputSet`, `TaskDocument`, and `Builder`
+
+# Introduction
+This tutorial will give you a comprehensive high-level overview of the key concepts in atomate2, diving into the important features of `Job` and `Flow` makers, as well as `InputSets`, `TaskDocuments`, and `Builders`.
+
+## `Job` and `Flow` makers
+
+`Job` and `Flow` makers are the workhorses of the atomate2 code framework. Their key role is to enable the user to orchestrate and execute a sequence of usually repetitive tasks, processes and manage computational workflows.
+To streamline the management of such processes, tools like [jobflow](https://github.com/materialsproject/jobflow) can be used. The two essential building blocks of the jobflow-based workflows in atomate2 are `Jobs` and `Flows`.
+
+### Basics
+
+A `Job` is a single computing job, and potentially can take any python function form, given that their inputs and outputs (return values) can be serialized in a JSON format. Jobs are defined using the `@job` decorator.
+
+A `Flow` is a sequential collection of job or other flow objects. The connectivity and also execution order and dependencies of the different jobs and flow is decided automatically from the job inputs. The output from one job (`job.output`) can be used as the input for the next one. This will therefore establish a connectivity between these two jobs.
+The connectivity between several jobs and flows can be arbitrary, depending on the purpose of the workflow. Flows are usually annotated with the `@dataclass` decorator.
```

Comment:
```suggestion
The connectivity between several jobs and flows can be arbitrary, depending on the purpose of the workflow.
```
I am a bit confused by the statement on the decorator here.

Reply:
Why are you confused? Do you think it belongs somewhere else in the text? Or do you mean that the @dataclass decorator is not a flow-specific decorated as contrary to the @job decorator?

Reply:
The `@dataclass` is used for the `Maker` but not for the Flow itself, right? 
I was preparing an update on my jobflow lecture and therefore looked at your draft ;). Just to explain why I revisited this.

Reply:
Oh yes, maybe I should clarify that it's used for the Maker. Thank you for seeing this :)

Reply:
I would just remove this sentence as you clearly describe it later for the Maker.